### PR TITLE
Feat: Backend - Data export Affective Annotation Project

### PR DESCRIPTION
**Pull request by haanh764**
This PR is covering the data exportation, and fixing the constraints in Scale model.

What's changed:
- `backend/data_export/models.py`: added new models for 2 type of exporting data (Scale exporting, and Text with question exporting) (2 new migrations - `backend/data_export/migrations/0002_exportedscale.py`, `backend/data_export/migrations/0003_exportedtextquestion.py`)
- `backend/data_export/pipeline/labels.py`: added 2 new types of exportation (Scale and Text with question)
- `backend/data_export/pipeline/formatters.py`: added `TupledTextFormatter` for formatting text with question as a tuple (`('text', 'question')`)
- `backend/data_export/pipeline/factories.py`: defined new factory for Affective Annotation data export, if the mode of project is `summary_mode` it will export label field only since there is no scaled question in this mode, if the mode of project is `emotions_mode` it will export scale field only since there is no textfield question in this mode, and for the others, humor, offensiveness mode, it will export both fields.
- `backend/data_export/pipeline/catalog.py`: removed the relation extraction in affective annotation project since we will not use it
- `backend/data_export/pipeline/examples/affective_annotation/_`: adjust the format of example files, removed example of relation extraction
- `backend/data_export/tests/test_task.py`: added test tasks for data export (there are 3 classes for 3 group of modes when export data - summary (export label only), emotions (export scale only), 3 other modes (export both label and scale))

For the fix in data_import:
- `backend/labels/models.py`: added contraint of unique between sample, user, and label
- `backend/labels/views.py`: adjust `ScaleListAPI` to delete the current scale if there is existed sample and label pair when creating a new scale value for that pair.

Screenshots of data_export:
Emotions mode (export scale only):
Screenshot from 2022-10-12 14-17-42
Summary mode (export label only)
Screenshot from 2022-10-12 14-21-05
Other modes (Others, Humor, Offensiveness - export both fields)
Screenshot from 2022-10-12 14-22-17

### CODE REVIEW

File: backend/labels/views.py [hunk 1]
```diff
@@ -118,6 +118,14 @@ class ScaleListAPI(BaseListAPI):
...
     serializer_class = ScaleSerializer
 
+    def create(self, request, *args, **kwargs):
+        if self.project.is_affective_annotation_project:
+            queryset = Scale.objects.filter(example=self.kwargs["example_id"], label=self.kwargs['label_id'])
+            print(queryset)
+            if not self.project.collaborative_annotation:
+                queryset = queryset.filter(user=self.request.user)
+            queryset.delete()
+        return super().create(request, args, kwargs)
```
Comment (teddy-f-47) [inline backend/labels/views.py]: Thanks for the quick fix! Unfortunately I have tried it and I got a key error from self.kwargs['label_id'].

When I tried to print kwargs, it only contains project id and user id. Meanwhile `request.data` contains something like this: `{'id': 0, 'label': 24, 'scale': 8, 'user': 0}` (The 'label' field contains the right label_id). So I think we need to adjust the code with something like: `label=request.data['label']`

Comment (teddy-f-47) [reply backend/labels/views.py]: I have tested it once more and I can confirm changing from `label=self.kwargs['label_id']` to `label=request.data['label']` fixes the issue :)

Comment (haanh764) [reply backend/labels/views.py]: ah yeah thank you for pointing it XD I forgot to check it XD updated

**Review by artikandri — APPROVED** [review]
Hello, I have connected the API and it works well, thank you for the good work! I also checked the code and everything looks good in my opinion, so approved  :)    But better to wait for Teddy's confirmation first.

**Review by teddy-f-47 — APPROVED** [review]
Hi, I have seen the code and everything looks fine. Thank you for the excellent work! I give my approval now.

However, about the fix for the bug that Argi showed today in the meeting, do you want to include it in this PR? I think we just need to adjust the code in `backend/labels/models.py`, for TextLabel to have `unique_together = ("example", "user", "text", "question")`, something like that 😅 Imo we can just add it in the next PR, but you can include it here if you want :)

Edit: Oh, I just saw you added the fix recently. So, great! Thank you once again XD

**Pull request merged**